commit 29cf7e8f54363de3ad614b3480517f02b84ade5e
Author: Nicolas Morey-Chaisemartin <nmoreychaisemartin@suse.com>
Date:   Thu May 2 09:56:40 2019 +0200

    mvapich2 fix double free
    
    Fix double free that causes a SEGV when mvapich2 fails to open
     a rdma device
    
    Signed-off-by: Nicolas Morey-Chaisemartin <nmoreychaisemartin@suse.com>

diff --git src/mpid/ch3/channels/mrail/src/gen2/rdma_iba_priv.c src/mpid/ch3/channels/mrail/src/gen2/rdma_iba_priv.c
index 89f666c8988b..48454297223c 100644
--- src/mpid/ch3/channels/mrail/src/gen2/rdma_iba_priv.c
+++ src/mpid/ch3/channels/mrail/src/gen2/rdma_iba_priv.c
@@ -790,10 +790,6 @@ int rdma_open_hca(struct mv2_MPIDI_CH3I_RDMA_Process_t *proc)
         }
 
         if (!ib_dev) {
-            /* Clean up before exit */
-            if (dev_list) {
-                ibv_free_device_list(dev_list);
-            }
             MPIR_ERR_SETFATALANDJUMP1(mpi_errno, MPI_ERR_OTHER, "**fail",
                                       "**fail %s", "No IB device found");
         }
